Harvest.Core/Services/SlothService.cs

```diff
@@ -0,0 +1,210 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Anlab.Core.Models.SlothModels;
+using Harvest.Core.Data;
+using Harvest.Core.Domain;
+using Harvest.Core.Models;
+using Harvest.Core.Models.Settings;
+using Harvest.Core.Models.SlothModels;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Serilog;
+
+namespace Harvest.Core.Services
+{
+    public interface ISlothService
+    {
+        Task<SlothResponseModel> MoveMoney(Transfer moneyTransfer);
+
+        Task ProcessTransferUpdates();
+    }
+
+    public class SlothService : ISlothService
+    {
+        private readonly AppDbContext _dbContext;
+        private readonly SlothSettings _slothSettings;
+        private readonly IFinancialService _financialService;
+
+        public SlothService(AppDbContext dbContext, IOptions<SlothSettings> slothSettings, IFinancialService financialService)
+        {
+            _dbContext = dbContext;
+            _slothSettings = slothSettings.Value;
+            _financialService = financialService;
+        }
+
+
+
+        //TODO: Add validation?
+        public async Task<SlothResponseModel> MoveMoney(Transfer moneyTransfer)
+        {
+            var token = _slothSettings.ApiKey;
+            var url = _slothSettings.ApiUrl;
+
+            if (string.IsNullOrWhiteSpace(token))
+            {
+                Log.Error("Sloth Token missing");
+            }
+
+            var debit = await _financialService.IsValid(moneyTransfer.FromAccount.Number);
+            var credit = await _financialService.IsValid(moneyTransfer.ToAccount.Number);
+
+            var model = new TransactionViewModel
+            {
+                MerchantTrackingNumber = moneyTransfer.Id.ToString(),
+                MerchantTrackingUrl = $"{_slothSettings.MerchantTrackingUrl}/{moneyTransfer.Id}"
+            };
+
+            model.Transfers.Add(new TransferViewModel
+            {
+                Account = moneyTransfer.FromAccount.Number?.ToUpperInvariant(),
```

Comment:
@sprucely If the account number is missing, this would be invalid.

Reply:
Also, maybe to be consistent, use the debit.KfsAccount like you do for chart and subaccount?

Reply:
This is equivalent to `SafeToUpper`, how it's done in Anlab... https://github.com/ucdavis/Anlab/blob/1786bab78194fcc618fae52ad18e99b486c1eb18/Anlab.Core/Services/SlothService.cs#L57